app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -55,6 +55,8 @@ def generic_callback(provider)
     if @user.persisted?
       sign_in_and_redirect @user, event: :authentication
       set_flash_message(:notice, :success, kind: provider.capitalize) if is_navigational_format?
+    elsif Flipper.enabled?(:block_registration)
+      redirect_to new_user_session_path, alert: "Registration is currently blocked"
```

Comment:
We need to make these texts i18n compatible. We can do that in the follow-up issue.

Reply:
sure i am opening one